src/test/CMakeLists.txt

```diff
@@ -13,16 +13,30 @@ include(ExternalProject)
 set(googletest_GIT_TAG "origin/cleaver")
 
 set(_build_byproducts)
-if(NOT WIN32)
+if(WIN32)
+  foreach(_lib IN ITEMS gtest gtest_main gmock gmock_main)
+    if("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
```

Comment:
Checking for `WIN32` and then comparing `CMAKE_BUILD_TYPE` does not seem valid. This should be revisited to check `CMAKE_CONFIGURATION_TYPES` and/or using generator expression

Reply:
Similarly to what was done below

Reply:
While I also tried a generator expression initially `BUILD_BYPRODUCTS` does not support generator expressions. For the *Ninja* generator, we use `CMAKE_BUILD_TYPE`.  I tested this patch with both the *Ninja* generator and the *Visual Studio* generator.

Reply:
Make sense, I suggest we do the conditional on the generator then.

Reply:
Because the current implementation assumes that `CMAKE_BUILD_TYPE` is defined on `WIN32` and this is not always the case.

Reply:
I suggest the following:
* introduce `_ep_arg_build_byproducts` internal var
* set it to `_ep_arg_build_byproducts BUILD_BYPRODUCTS "${_build_byproducts}"` only if the generator is Ninja or if a single configuration generator

Reply:
Done